comp/process/agent/agentimpl/agent.go

```diff
@@ -70,37 +70,21 @@ func newProcessAgent(p processAgentParams) optional.Option[agent.Component] {
 	// Look to see if any checks are enabled, if not, return since the agent doesn't need to be enabled.
 	if len(enabledChecks) == 0 {
 		p.Log.Info(agentDisabledMessage)
-		return optional.NewNoneOption[agent.Component]()
+		return processAgent{
+			enabled: false,
+		}
 	}
 
 	processAgentComponent := processAgent{
-		Checks: enabledChecks,
-		Log:    p.Log,
-		Runner: p.Runner,
-	}
-
-	p.Lc.Append(fx.Hook{
-		OnStart: processAgentComponent.start,
-		OnStop:  processAgentComponent.stop,
-	})
-
-	return optional.NewOption[agent.Component](processAgentComponent)
-}
-
-func (p processAgent) start(ctx context.Context) error {
-	p.Log.Info("process-agent starting")
-
-	chks := make([]string, 0, len(p.Checks))
-	for _, check := range p.Checks {
-		chks = append(chks, check.Name())
+		enabled: true,
+		Checks:  enabledChecks,
+		Log:     p.Log,
 	}
-	p.Log.Info("process-agent checks", log.Object("checks", chks))
 
-	return p.Runner.Run(ctx)
+	return processAgentComponent
 }
 
-func (p processAgent) stop(_ context.Context) error {
-	p.Log.Info("process-agent stopping")
-
-	return nil
+// Enabled determines whether the process agent is enabled based on the configuration.
+func (p processAgent) Enabled() bool {
```

Comment:
Having `Enabled` is fine but I am wondering the difference with using optinal.Option[Component]?

Reply:
I spoke with @hush-hush about this yesterday and my understanding is that for logs-agent components it was optional as it may not be available in a given environment/compilation. The usage of optional may be better in that context. 

Also from the process-agent's main_common the `Enabled` method has better overall readability then using optional.